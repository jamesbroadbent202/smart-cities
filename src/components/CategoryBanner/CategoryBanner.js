import React from 'react';
import PropTypes from 'prop-types';
import IndicatorCard from '../Card/IndicatorCard/IndicatorCard';
import { INDICATORS } from '../../constants';
import aggregateIndicatorForCities from '../../helpers/aggregateIndicatorForCities';
import getColorVariant from '../../helpers/getColorVariant';
import style from './CategoryBanner.scss';

const CategoryBanner = (props) => {
  const backgroundColor = getColorVariant(props.category.colorName, '020');
  const cardHighlightColor = getColorVariant(props.category.colorName, '500');

  return (
    <div style={{ backgroundColor }}>
      <div className={style.container}>
        <div className={style.titleText}>
          <h1 className={style.title}>{props.category.name}</h1>

          <p className={style.description}>{props.category.description}</p>
        </div>

        <IndicatorCard
          size="large"
          className={style.card}
          color={cardHighlightColor}
          indicator={props.category.heroIndicatorId}
          value={aggregateIndicatorForCities(
            props.category.heroIndicatorId,
            props.cities,
          )}
        />
      </div>
    </div>
  );
};

CategoryBanner.propTypes = {
  category: PropTypes.shape({
    colorName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    heroIndicatorId: PropTypes.oneOf(Object.keys(INDICATORS)).isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    indices: PropTypes.object.isRequired,
  })).isRequired,
};

export default CategoryBanner;
