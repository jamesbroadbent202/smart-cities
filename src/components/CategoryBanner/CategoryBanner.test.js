import React from 'react';
import { shallow } from 'enzyme';
import CategoryBanner from './CategoryBanner';

jest.mock('../Card/IndicatorCard/IndicatorCard', () => 'IndicatorCard');

const defaultProps = {
  category: {
    colorName: 'jobs',
    description: 'This is the mock category',
    heroIndicatorId: 'population',
    id: 'mock',
    name: 'Mock category',
  },
  cities: [
    {
      name: 'Perth',
      indices: {}
    },
    {
      name: 'Sydney',
      indices: {}
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <CategoryBanner {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should set the background color', () => {
  const component = shallow(
    <CategoryBanner {...defaultProps} />
  );

  const componentStyle = component.find('div').at(0).prop('style');
  // note, here the color is the string 'JOBS_020' because of the way
  // styles are imported during tests, but in reality
  // this will be the hex color
  expect(componentStyle.backgroundColor).toBe('JOBS_020');
});
