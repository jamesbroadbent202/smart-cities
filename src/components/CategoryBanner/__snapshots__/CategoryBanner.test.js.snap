// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div style={{...}}>
  <div className=\\"container\\">
    <div className=\\"titleText\\">
      <h1 className=\\"title\\">
        Mock category
      </h1>
      <p className=\\"description\\">
        This is the mock category
      </p>
    </div>
    <IndicatorCard size=\\"large\\" className=\\"card\\" color=\\"JOBS_500\\" indicator=\\"population\\" value={0} />
  </div>
</div>"
`;
